File: practice/easy/0070-Climbing_Stairs.py
class Solution:
  def climbStairs(self, n: int) -> int:
    ways = [0, 1, 2]
    if n < 3:
      return ways[n]

    for i in range(3, n+1):
      ways.append(ways[i-1] + ways[i-2])

    return ways[-1]

File: practice/csharp/easy/0070_ClimbingStairs.cs
public class Solution {
  public int ClimbStairs(int n) {
    // for the nth stair, add the number of ways to get to n-1 and n-2
    // add the first 2 solutions for simplicity

    if (n < 3) {
      return n;
    }

    int[] stairs = new int[n];
    stairs[0] = 1;
    stairs[1] = 2;
    
    for (int i = 2; i < n; i++) {
      stairs[i] = stairs[i-1] + stairs[i-2];
    }

    return stairs.Last();
  }
}
